dmutils/logging.py

```diff
@@ -138,7 +138,7 @@ def format(self, record):
         try:
             record.msg = record.msg.format(**record.__dict__)
         except KeyError:
-            logger.error("failed to format log message")
+            logger.exception("failed to format log message")
```

Comment:
Is there any information we can get from the KeyError to add to the message here - e.g. what was the key that broke?